src/viur/core/modules/user.py

```diff
@@ -1452,6 +1488,60 @@ def authenticateUser(self, key: db.Key, **kwargs):
 
         return self.render.loginSucceeded(**kwargs)
 
+    # Action for primary authentication selection
+
+    def SelectAuthenticationProviderSkel(self):
+        providers = {}
+        for provider in self.authenticationProviders:
+            provider = getattr(self, f"auth_{provider.__name__.lower()}")
+            providers[provider.start_url] = provider.NAME
+
+        class SelectAuthenticationProviderSkel(skeleton.RelSkel):
+            provider = SelectBone(
+                descr="Authentication method",
+                required=True,
+                values=providers,
+            )
+
+        return SelectAuthenticationProviderSkel()
+
+    @exposed
+    def selectauthenticationprovider(self, **kwargs):
```

Comment:
```suggestion
    def select_authentication_provider(self, **kwargs):
```
I find this version cleaner what do you think?

Reply:
we can do it. The reason why I chose this name is, that a "selectauthenticationprovider"-action will return a "selectauthenticationproviderSuccess" (for legacy reasons). A "select_authentication_provider" will return "select_authentication_providerSuccess", which looks a little bit dumb.